src/error.rs

```diff
@@ -66,9 +66,8 @@ quick_error! {
             from()
         }
         /// Error preparing connection info.
-        PrepareConnectionInfo(e: io::Error) {
```

Comment:
Why was the error removed? It looks like (futher down) the error that should be getting wrapped here just gets discarded.

Reply:
Yeah, I just remove the `io:Error parameter`, because unbonded channel doesn't produce this error: ```
        ch1.into_future()
            .and_then(move |(msg_opt, chann)| {
                Ok(PrivConnectionInfo {
                    id: our_uid,
                    for_direct: direct_addrs.into_iter().collect(),
                    for_hole_punch: vec![],
                    hole_punch_socket: None,
                    p2p_conn_info: unwrap!(msg_opt),
                    rendezvous_channel: chann,
                    connection_rx: conn_rx,
                })
            })
            .map_err(|_| CrustError::PrepareConnectionInfo)
            .into_boxed()
```

Reply:
What error is getting produced here but getting discarded? If it's a `Void` then we don't need `CrustError::PrepareConnectInfo` at all and we can just use `.infallible()`. If it's a `()` then write `.map_err(|()| CrustError::PrepareConnectionInfo)`. If it's anything else then that should be the error type carried by `PrepareConnectionInfo`.